src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Performance/CSharpPreferDictionaryTryMethodsOverContainsKeyGuardFixer.cs

```diff
@@ -46,12 +62,12 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
                 var node = root.FindNode(location.SourceSpan, getInnermostNodeForTie: true);
                 switch (node)
                 {
-                    case ElementAccessExpressionSyntax:
+                    case ElementAccessExpressionSyntax when diagnostic.Id == PreferDictionaryTryMethodsOverContainsKeyGuardAnalyzer.PreferTryGetValueRuleId:
```

Comment:
Seems this method only called when `diagnostic.Id == PreferDictionaryTryMethodsOverContainsKeyGuardAnalyzer.PreferTryGetValueRuleId` is true, why do we need to add this condition here again?
```suggestion
                    case ElementAccessExpressionSyntax :
```